src/plugins/intel_gpu/src/kernel_selector/cl_kernels/normalize_gpu_within_spatial_ref.cl

```diff
@@ -13,21 +13,45 @@ KERNEL (normalize_gpu_within_spatial_ref)(
     const __global SCALE_TABLE_TYPE* scale_input
 )
 {
-    const uint x = get_global_id(0);
-    const uint y = get_global_id(1);
+    const uint gws0 = get_global_id(0);
+    const uint gws1 = get_global_id(1);
     const uint b = get_global_id(2);
+    uint f, y, x = 0;
 
     // Compute norm
     uint input_idx;
     float norm = EPSILON;
-    for (int i = 0; i < INPUT0_FEATURE_NUM; i++)
+#if NORM_AXIS == 1
+    for (f = 0; f < INPUT0_FEATURE_NUM; f++)
     {
-        input_idx = INPUT0_GET_INDEX(b, i , y, x);
+        x = gws0;
```

Comment:
Why is it being allocated every time? I think it would be enough to set it just once outside the for loop.

Reply:
Because allocated value is different per NORM_AXIS. With different f, y, x, calculated norm is also changed.